src/components/Todo/EditCard/edit-card.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+import EditActionBar from "components/Todo/EditCard/edit-action-bar.component";
+import { ENTER } from "common/constants";
+
+const EditCard = ({
+  todo,
+  handleEditTask,
+  handleCompleteTask,
+  handleEditCancelTask,
+}) => {
+  const { id, task } = todo;
+  const textRef = useRef(null);
+  const [editedTask, setEditData] = useState(task);
+
+  const handleChange = (e) => {
+    setEditData(e.target.value);
+  };
+  const handleKeyDown = (e) => {
+    if (e.key === ENTER) {
+      handleEditTask(id, editedTask);
+    }
+  };
+
+  const setFocusAndSelection = () => {
```

Comment:
see https://github.com/azo-wtag/todo-app-react/pull/4/files#r1172078490

```suggestion
  function setFocusAndSelection() {
```